### PR TITLE
LibGfx+icc: Look at profile_size field 

### PR DESCRIPTION
**Pull request by nico**
This trims the input bytes to the profile size stored in the file.
Alternatively. we could reject files where the stored size doesn't
match the handed in size. But ICC profiles can be embedded in other
files, and those could conceivably pad the ICC profile data some.

---

With this, ICC profile header processing is done! A breezy 940 lines to process 128 bytes.

### CODE REVIEW

File: Userland/Libraries/LibGfx/ICCProfile.cpp [hunk 1]
```diff
@@ -116,6 +116,22 @@ struct ICCHeader {
...
 static_assert(sizeof(ICCHeader) == 128);
 
+ErrorOr<u32> parse_size(ICCHeader const& header, ReadonlyBytes icc_bytes)
+{
+    // ICC v4, 7.2.2 Profile size field
+    // "The value in the profile size field shall be the exact size obtained by combining the profile header,
+    // the tag table, and the tagged element data, including the pad bytes for the last tag."
+
+    // Valid files have enough data for profile header and tag table entry count.
+    if (header.profile_size < sizeof(ICCHeader) + sizeof(uint32_t))
```
Comment (linusg) [inline Userland/Libraries/LibGfx/ICCProfile.cpp:126]: Can we use u32 here instead?

File: Userland/Libraries/LibGfx/ICCProfile.h [hunk 2]
```diff
@@ -237,6 +238,7 @@ class Profile : public RefCounted<Profile> {
     static Crypto::Hash::MD5::DigestType compute_id(ReadonlyBytes);
 
 private:
+    u32 m_on_disk_size;
```
Comment (linusg) [inline Userland/Libraries/LibGfx/ICCProfile.h:241]: Please explicitly initialize primitive members

**Review by linusg — COMMENTED** [review]
`Alternatively.` -> `Alternatively,` in commit message, probably

**Comment by nico** [comment]
Good comments, thanks. All done.

**Pull request merged**